### PR TITLE
Updates for RestResponse compatibility

### PR DESCRIPTION
**Pull request by RikkiGibson**
- need to bump minor

### CODE REVIEW

**Comment by RikkiGibson** [comment]
nyc is broken in node 6. I'm sure it'll sort itself out when we do a proper publish of ms-rest-js.

File: package.json [hunk 1]
```diff
@@ -30,13 +30,13 @@
...
   "license": "MIT",
   "dependencies": {
-    "ms-rest-js": "~0.20.403",
     "tslib": "^1.9.2"
   },
   "devDependencies": {
     "@types/mocha": "^5.2.1",
     "@types/should": "^8.1.30",
     "mocha": "^5.2.0",
+    "ms-rest-js": "github:azure/ms-rest-js#response-flattening",
```
Comment (daschult) [inline package.json]: Make sure to change this back to the preview version.

Comment (daschult) [reply package.json]: And change it back to being a normal dependency too.

File: lib/lroPoller.ts [hunk 2]
```diff
@@ -105,15 +105,24 @@ export class LROPoller {
...
+      result = Promise.resolve(flattenResponse(initialResponse, getter && operationSpec && getter(operationSpec, initialResponse)));
     } else {
       result = lroPollStrategy.pollUntilFinished().then((succeeded: boolean) => {
         if (succeeded) {
-          return lroPollStrategy.getOperationResponse();
+          return lroPollStrategy.getOperationResponse().then(res => {
+            const response = this.getMostRecentResponse();
+            const getter = response.request.operationResponseGetter;
+            const responseSpec = getter && operationSpec && getter(operationSpec, response);
+            return flattenResponse(res, responseSpec);
```
Comment (daschult) [inline lib/lroPoller.ts]: The previous three lines are functionally equivalent to lines 115 and 116. Can you extract them into a method that takes an `HttpOperationResponse` and returns a `Promise<RestResponse>`?

Comment (RikkiGibson) [reply lib/lroPoller.ts]: On it

Comment (RikkiGibson) [reply lib/lroPoller.ts]: Regarding line 121: Is there ever going to be a difference between using the `res` and accessing `getMostRecentResponse`? Shouldn't I just use `res` here?

**Comment by RikkiGibson** [comment]
All good @daschult?

**Pull request merged**